app/models/taken_test.rb

```diff
@@ -1,4 +1,51 @@
 class TakenTest < ApplicationRecord
   belongs_to :test
   belongs_to :user
+  belongs_to :current_question, class_name: 'Question', optional: true
+
+  before_validation :before_validation_set_current_question
+
+  def accept!(answer_ids)
+    answer_ids.shift
+    self.correct_questions += 1 if correct_answer?(answer_ids)
+    save!
+  end
+
+  def completed?
+    current_question.nil?
+  end
+
+  def successful?
+    correct_questions / test.questions.size >= 0.85
```

Comment:
магические цифры правильно выносить в константы
например: 
```suggestion
    correct_questions / test.questions.size >= MINIMUM_PERCENT
```